### PR TITLE
fixes #929 : Use Gmail api to fetch user email aliases

### LINKED ISSUE #929: use Gmail api for aliases
https://developers.google.com/gmail/api/v1/reference/users/settings/sendAs/get

### PR DESCRIPTION
**Pull request by vincentp**
Anything that needs to be done to setup tests so I can write some? I get:
`Error: ENOENT: no such file or directory, open 'test/test-secrets.json'`

Also, we could filter out emails on "verificationStatus" (user might not have validated his alias) or leave that to users to decide. What do you think?

### CODE REVIEW

File: extension/js/common/settings.ts [hunk 1]
```diff
@@ -21,18 +21,13 @@ export class Settings {
...
-        return results.filter(email => !Value.is(email).in(Settings.ignoreEmailAliases));
-      }
-      for (const from of headers.from) {
-        results.push(Str.parseEmail(from).email);
-        query += ' -from:"' + Str.parseEmail(from).email + '"';
-      }
-    }
+    const response = await Google.gmail.fetchAcctAliases(acctEmail);
+
+    const aliases = response.sendAs.map(alias => {
```
Comment (tomholub) [inline extension/js/common/settings.ts]: can shorten to to `map(alias => alias.sendAsEmail)`

Also can remove the empty lines

File: extension/js/common/api/api.ts [hunk 2]
```diff
@@ -194,6 +194,8 @@ export namespace R { // responses
   export type GmailDraftUpdate = {};
   export type GmailDraftGet = { id: string, message: GmailMsg };
   export type GmailDraftSend = {};
+  export type GmailAliases = { sendAs: GmailAlias[] };
+  export type GmailAlias = { sendAsEmail: string, displayName: string, replyToAddress: string, signature: string, isDefault: boolean, treatAsAlias: boolean, verificationStatus: string };
```
Comment (tomholub) [inline extension/js/common/api/api.ts]: Looks good but don't export `GmailAlias`, only `GmailAliases`. Actually, I have a convention for this: I recommend to rename `GmailAlias` to `GmailAliases$sendAs`.

File: extension/js/common/settings.ts [hunk 3]
```diff
@@ -21,18 +21,13 @@ export class Settings {
...
+      return alias.sendAsEmail;
+    });
+
+    return aliases.filter(email => !Value.is(email).in(Settings.ignoreEmailAliases));
```
Comment (tomholub) [inline extension/js/common/settings.ts]: Actually you could probably remove `ignoreEmailAliases` completely (not needed anymore if we get aliases from authoritative source) and instead start filtering on whether aliases were verified as you mentioned.

**Comment by tomholub** [comment]
As for the tests, you will have to wait for our CI tests to pick up this branch (I'll allow it manually). You won't be able to run them locally due to the missing file.

**Comment by tomholub** [comment]
@vincentp Please confirm that you release this (and any following) contributions under MIT so that I can merge them. Thanks!

**Comment by vincentp** [comment]
MIT confirmed, thanks for the feedback :)

**Comment by tomholub** [comment]
This was not bad. If you want to take a look at other issues marked as `actionable`, we should probably start paying you. How long did this take?

**Comment by tomholub** [comment]
Minor feedback: `verificationStatus` type should ideally be union of literals, eg `'accepted' | 'new' | 'whatever'` rather than just a `string`.

**Pull request merged**